### ISSUE TITLE
Consider the final path segment in url, if not terminated by a separator, to be the variant

### ISSUE THREAD

**Issue by ansongoldade** [lynx-json/lynx-docs#78]
It would be nice to be able to use a URI like `http://host/realm1/realm2/default-something` as the `action` for a `submit` and have lynx-docs resolve the `default-something` variant for realm `/realm1/reaml2/`. This would allow modeling application flow to variants other than `default` as the result of invoking submit controls. I believe the only way to accomplish this currently is using a JavaScript variant as the `default` variant or including an input in the form being submitted with the name `variant` and making the method a `GET`.

**Comment by johnhowes** [comment]
This seems like a good idea.

**Comment by danmork** [comment]
Currently, for a `method` of `POST`, you can set the `submit` `action` URI to `?variant=index` or `?variant=non-default-variant` to get to the next resource. The user agent isn't stripping off the query component of the `action` so it works. I'm not sure if that behavior is correct or not – we'd need to compare with HTML to be sure. However, it is at least a workaround until the behavior changes.

**Comment by ansongoldade** [comment]
Based on the HTML and Lynx form submission specs the user agent is handing submissions properly.
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#form-submission-2
http://lynx-json.org/specification/processes/submitting_form_data.html

The query string component of the `action` is removed and substituted with the result of encoding the form data set for `GET` and `DELETE` submissions. That is where this functionality may provide some value. Since the problem in front of me is solved with `POST` submissions we can let a future use case determine if or when we want to further consider this.

**Comment by johnhowes** [comment]
I don't read this as "the user agent is handling submissions properly". Dan said that the user agent is not stripping off the querystring value for POSTs, but the section of the lynx spec you referenced says that "Let the URL's query component be empty." for POSTs, PUTs, or other methods that support an entity body. I would say that the user agent is not currently compliant. Right?

**Comment by ansongoldade** [comment]
You're correct. I missed that bullet point in the POST section of the Lynx specification. I would agree that the user agent is out of compliance with the Lynx specification. However, its current behavior is in alignment with the way HTML forms work. Does anyone recall the reason we diverged from HTML on this point?
